Plugins/Mono/Mono_Handlers.cpp

```diff
@@ -396,4 +400,31 @@ int32_t ClosureActionDoCommand(uint32_t oid, uint64_t eventId)
     return 0;
 }
 
+std::shared_ptr<char> ISO88959ToUTF8(const char *str)
+{
+    char* utf8 = new char[1 + (2 * strlen(str))];
+
+    int len = 0;
+
+    if (utf8) {
+        char *c = utf8;
+        for (; *str; ++str) {
+            if (!(*str & 0x80)) {
+                *c++ = *str;
+                len++;
+            } else {
+                *c++ = (char) (0xc2 | ((unsigned char)(*str) >> 6));
+
+                *c++ = (char) (0x3f & *str);
+                len += 2;
+            }
+        }
+        *c++ = '\0';
+    }
+
+    realloc(utf8, len);
```

Comment:
Mixing `realloc` with `new` is not a good idea. glibc implements them both with the same allocator, so it will work in most cases, but that can change between versions too. Just use `new`.. If the memory usage is problematic, you'll need a secondary `new`'d buffer, then manually memmove the data over.
But I think a wee bit of extra usage is no big deal.